pyproject.toml

```
@@ -17,8 +17,32 @@ classifiers = [
     "Operating System :: POSIX",
 ]
 requires-python = ">=3.11"
-# Use requirements/main.in for runtime dependencies instead.
-dependencies = []
+dependencies = [
+    "idna>=3.7",
+    "setuptools>70.0",
+    "certifi>2024.2.0",
+    "pillow>=10.3.0",
+    "ecdsa>=0.18",
+    "BeautifulSoup4",
+    "click",
+    "jinja2>=3.1.4",
+    "marshmallow",
+    "pandoc",
+    "pypandoc",
+    "requests",
+    "setuptools_scm",
+    "tox",
+    "pre-commit",
+    "mypy",
+    "pydantic",
+    "requests>2.32.0",
+    "arrow",
+    "urllib3 >=2.2.2",
+    "tlslite-ng",
+    "oauth2",
+    "types-requests",
+    "zephyr-python-api"
+]
```

I think this is closer to what you need. I'd have to look more closely if you need direct dependencies on things like `certifi` and `idna` because they often come through requests.

```suggestion
dependencies = [
    "idna>=3.7",
    "certifi>2024.2.0",
    "pillow>=10.3.0",
    "ecdsa>=0.18",
    "BeautifulSoup4",
    "click",
    "jinja2>=3.1.4",
    "marshmallow",
    "pandoc",
    "pypandoc",
    "pydantic",
    "requests>2.32.0",
    "arrow",
    "urllib3 >=2.2.2",
    "tlslite-ng",
    "oauth2",
    "zephyr-python-api"
]
```

The other dependencies you can keep in a `requirements/dev.in` / dev.txt file that gets installed in GitHub Actions and developer environments.